### ISSUE TITLE
Enable Pipe Delimited Search (Apryse)

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#4614]
* As an IAO Analyst
* I want to be able to search multiple keywords at once
* so that I can more efficiently search a document

**Assumptions & Scope**	
This story is related to the legal keyword search. We didn't realize that the pipe delimited search was something that needed to be configured, and when we rolled it back we lost the functionality. This ticket is to re-enable it for the document reviewer.

What is IN scope?
allowing users to create their own pipe delimited searches in Apryse

What is NOT in scope?
NO preset terms for searching

**Acceptance Criteria**	

**Scenario 1: Find Text - keywords**
* GIVEN an IAO user is on the find text feature of the doc reviewer
* WHEN they enter a keyword
* THEN the results will begin populating

**Scenario 2: Find Text - Add Keyword**
* GIVEN an IAO user has entered a word in the find text of the doc reviewer
* WHEN they add a pipe ( | ) 
* AND follow it with another word
* THEN the original keyword, and the newly added word will be searched for in the document

**Scenario 3: Find Text - Multiple pipes**
* GIVEN an IAO user is on the find text feature of the doc review
* WHEN they add multiple words separated by pipes
* THEN all keywords separated by pipes will be searched

**Scenario 4: Find and Redact - keywords**
* GIVEN an IAO user is on the find and redact feature of the doc reviewer
* WHEN they enter a keyword
* THEN the results will begin populating

**Scenario 5: Find and Redact - Add Keyword**
* GIVEN an IAO user has entered a word in the find and redact feature of the doc reviewer
* WHEN they add a pipe ( | ) 
* AND follow it with another word
* THEN the original keyword, and the newly added word will be searched for in the document

**Scenario 6: Find and Redact - Multiple pipes**
* GIVEN an IAO user is on the find and redact feature of the doc review
* WHEN they add multiple words separated by pipes
* THEN all keywords separated by pipes will be searched

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by nimya-aot** [comment]
When we search a single word on redaction panel,  same result comes up two times and if user accidently select both of them and click 'Add Mark' then two redactions are generating on the page. 
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOWVhQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--46f67af4996ca181b6d055321ad07b742f207935/MicrosoftTeams-image%20%281%29.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOWlhQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--017db4b59cf8daa6cfd6d31e1123e9458935964d/image.png)

@Aman-Hundal Needs to fix this

**Comment by Aman-Hundal** [comment]
@nimya-aot Hey Nimya, I removed my code I added for this ticket and the issue you mentioned still occurs, even with the prior code we had. I created a bug ticket [here](https://app.zenhub.com/workspaces/foi-flow-607f0ab6ae54b700111dc733/issues/gh/bcgov/foi-flow/4670), as I noticed this before, I dont believe its a stopper for 9.0 (will confirm with biz / dev) but its definetly something I will look into. 

However as its unrelated to this ticket as the issue exited prior, should we move this ticket back to QA?
